src/test/java/uk/gov/companieshouse/chs/gov/uk/notify/integration/api/letterdispatcher/LetterDispatcherTest.java

```diff
@@ -0,0 +1,124 @@
+import uk.gov.companieshouse.chs.gov.uk.notify.integration.api.service.GovUkNotifyService;
+import uk.gov.companieshouse.chs.gov.uk.notify.integration.api.service.Postage;
+import uk.gov.companieshouse.chs.gov.uk.notify.integration.api.templatelookup.LetterTemplateKey;
+import uk.gov.companieshouse.chs.gov.uk.notify.integration.api.templatepersonalisation.TemplatePersonaliser;
+import uk.gov.companieshouse.chs.gov.uk.notify.integration.api.utils.PersonalisationDetailsParser;
+import uk.gov.companieshouse.logging.Logger;
+
+@ExtendWith(MockitoExtension.class)
+@Tag("unit-test")
+class LetterDispatcherTest {
+
+    @Mock
+    private GovUkNotifyService govUkNotifyService;
+    @Mock
+    private NotificationDatabaseService notificationDatabaseService;
+    @Mock
+    private TemplatePersonaliser templatePersonaliser;
+    @Mock
+    private HtmlPdfGenerator pdfGenerator;
+    @Mock
+    private PersonalisationDetailsParser parser;
+    @Mock
+    private Logger logger;
+
+    @InjectMocks
+    private LetterDispatcher letterDispatcher;
+
+    @Test
+    void testSendLetter_success() throws IOException {
+        // Arrange
+        Postage postage = Postage.FIRST_CLASS;
+        String reference = "ref";
+        String appId = "app";
+        String letterId = "letter";
+        String templateId = "template";
+        Address address = new Address();
+        String contextId = "ctx";
+        String personalisedLetter = "<html>letter</html>";
+        String govNotifyReference = appId + "-" + letterId + "-" + reference;
+        String personalisationDetails = "{}";
+        Map<String, String> personalisationDetailsMap = Collections.emptyMap();
+
+        when(parser.parsePersonalisationDetails(personalisationDetails, contextId))
+                .thenReturn(personalisationDetailsMap);
+        when(templatePersonaliser.personaliseLetterTemplate(
+                new LetterTemplateKey(appId, letterId, templateId), reference,
+                personalisationDetailsMap, address)).thenReturn(personalisedLetter);
+
+        InputStream pdfStream = new ByteArrayInputStream(new byte[] {});
+        when(pdfGenerator.generatePdfFromHtml(personalisedLetter, govNotifyReference))
+                .thenReturn(pdfStream);
+
+        GovUkNotifyService.LetterResp letterResp = mock(GovUkNotifyService.LetterResp.class);
+        when(govUkNotifyService.sendLetter(postage, govNotifyReference, pdfStream))
+                .thenReturn(letterResp);
+
+        // Act
+        GovUkNotifyService.LetterResp result = letterDispatcher.sendLetter(postage, reference,
+                appId, letterId, templateId, address, personalisationDetails, contextId);
+
+        // Assert
+        assertEquals(letterResp, result);
+        verify(notificationDatabaseService).storeResponse(letterResp);
+    }
+
+    @Test
+    void testSendOldLetter_success() throws IOException {
+        // Arrange
+        Postage postage = Postage.ECONOMY;
```

Comment:
These variables has also been used in the previous test. Could we define all these duplicated local variables as class constants ?